client/client.go

```diff
@@ -282,15 +250,27 @@ func (c *Client) withAccountIDRegionAndNamespace(accountID, region, namespace st
 	}
 }
 
+func validateRegions(logger hclog.Logger, regions []string) (bool, error) {
+	// validate regions values
+	for i, region := range regions {
+		if i != 0 && region == "*" {
+			return false, fmt.Errorf("region wildcard \"*\" is only supported as first argument")
+		}
+	}
+
+	var wildcardAllRegions bool
+	if len(regions) == 1 && regions[0] == "*" || len(regions) == 0 {
+		logger.Info("All regions specified in config.yml. Assuming all regions")
```

Comment:
I think we can remove the dependency for the logger here and move this to an outside of this function. This way this function is more pure and can be more easily re-used in other-places.

Reply:
fixed